### ISSUE TITLE
providers/consul: support setting ID on consul_service

### ISSUE THREAD

**Issue by dmportella** [hashicorp/terraform#9352]
Hi there,

Right now the resource `consul_service` *does not* support the resource count features very well because since the consul record identifier is created from the name you give to the record ([Reference](https://www.terraform.io/docs/providers/consul/r/service.html#id)) you can't create record with the same `name` but different `identifiers` which allows you to use `consul` dns load balancer feature.

### Terraform Version
> Terraform v0.7.5

### Affected Resource(s)
- consul_service

### Terraform Configuration Files
```
# Docker provider
provider "docker" {
	host = "unix:///var/run/docker.sock"
}

# Consul Provider
provider "consul" {
	address	= "http://localhost:8500"
	datacenter = "dc1"
	scheme	 = "http"
}

variable "elastic_count" {
	type = "string"
	default = "4"
}

resource "docker_image" "elastic" {
	name = "elasticsearch:2.4.1"
}

# Create a container
resource "docker_container" "elastic" {
	count = "${var.elastic_count}"
	image = "${docker_image.elastic.name}"
	name = "elastic-${format("%02d", count.index+1)}"
	
	hostname = "elastic-${format("%02d", count.index+1)}"

	restart = "always"
	memory = 512

	dns = ["172.17.0.1"]

	labels {
		type = "elastic"
	}

	volumes {
		container_path  = "/usr/share/elasticsearch/data"
		host_path = "/home/dmportella/_workspaces/terraform/consul/data/elastic/data.${format("%d", count.index+1)}"
		read_only = false
	}

	volumes {
		container_path  = "/usr/share/elasticsearch/config"
		host_path = "/home/dmportella/_workspaces/terraform/consul/configs/elastic"
		read_only = false
	}
}

resource "consul_service" "elastic-single" {
    depends_on = ["null_resource.consul_provisioned"]
    count = "${var.elastic_count}"

    address = "${element(docker_container.elastic.*.ip_address, count.index)}"
    name = "elastic-${format("%02d", count.index+1)}"
    port = 9200
    tags = ["elastic"]
}

resource "consul_service" "elastic-cluster" {
    depends_on = ["null_resource.consul_provisioned"]
    count = "${var.elastic_count}"

    id = "elastic-cluster-${format("%02d", count.index+1)}"
    address = "${element(docker_container.elastic.*.ip_address, count.index)}"

    name = "elastic"
    port = 9200
    tags = ["elastic", "cluster"]
}
```
The resource `consul_service.elastic-cluster` should create 4 entries in consul all with the same name but with different `identifiers`. However it creates only one with the last address of the array of resources.

See below the consul `http://localhost:8500/v1/agent/services` http api request after the terraform apply is run.
```
{
  "consul": {
    "ID": "consul",
    "Service": "consul",
    "Tags": [],
    "Address": "",
    "Port": 8300,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "elastic": {
    "ID": "elastic",
    "Service": "elastic",
    "Tags": [
      "elastic",
      "cluster"
    ],
    "Address": "172.17.0.9",
    "Port": 9200,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "elastic-01": {
    "ID": "elastic-01",
    "Service": "elastic-01",
    "Tags": [
      "elastic"
    ],
    "Address": "172.17.0.10",
    "Port": 9200,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "elastic-02": {
    "ID": "elastic-02",
    "Service": "elastic-02",
    "Tags": [
      "elastic"
    ],
    "Address": "172.17.0.8",
    "Port": 9200,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "elastic-03": {
    "ID": "elastic-03",
    "Service": "elastic-03",
    "Tags": [
      "elastic"
    ],
    "Address": "172.17.0.7",
    "Port": 9200,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "elastic-04": {
    "ID": "elastic-04",
    "Service": "elastic-04",
    "Tags": [
      "elastic"
    ],
    "Address": "172.17.0.9",
    "Port": 9200,
    "EnableTagOverride": false,
    "CreateIndex": 0,
    "ModifyIndex": 0
  }
}
```

### Debug Output
Application errors on validation no debug logs.

### Panic Output
Application errors on validation no panic logs.

### Expected Behavior
What should have happened?

It should create 4 entries in consul each with different `identifiers` but with the same `name`

### Actual Behavior
What actually happened?

Application errors on validation with: 

```
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * consul_service.elastic-cluster #0: "id": this field cannot be set
  * consul_service.elastic-cluster #1: "id": this field cannot be set
  * consul_service.elastic-cluster #2: "id": this field cannot be set
  * consul_service.elastic-cluster #3: "id": this field cannot be set

```

### Steps to Reproduce

You need a consul cluster available I have used docker to test my setups so the consul docker image is very a very good start. Once it is there use terraform apply to see the error.

### Important Factoids

This is a feature request but could be considered a design issue as one would expect it be able to support the dns load balancing feature of consul which requires the user to pass in records with the same `name` but different `identifiers`.

**Comment by dmportella** [comment]
I would be happy to make the changes to the source code just wanted to raise the issue so it could be discussed.

**Comment by dmportella** [comment]
While trying to implement a patch for the above I found some real weird behaviour. Doesn't matter if you change the `id` be set-able terraform is always using the name field as the I'd of the resources. It only works if I rename the `id` field to something else like identifier

**Comment by dmportella** [comment]
Tomorrow I will get a pull request in so people can see what I mean. I tried to implement so a user can set the I'd to whatever they want. I made it optional so the previous setup would still work as I didn't want to introduce a breaking change.

**Comment by stack72** [comment]
Closed via #9366
